pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -570,6 +572,107 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param maxlength the max length of the prefix strings for the string.
+   * @return generate an array of prefix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] uniquePrefixes(String input, int maxlength) {
+    ObjectSet<String> prefixSet = new ObjectLinkedOpenHashSet<>();
```

Comment:
We don't need a set since all prefixes have different length. Same for other functions

Reply:
updated, thanks